src/legal-officer/client.ts

```diff
@@ -343,9 +343,16 @@ export async function vote(params: {
     signer: Signer,
     callback: SignCallback,
 }): Promise<Vote> {
-    const { client, vote, myVote, callback } = params;
+    const { client, vote, myVote, signer, callback } = params;
+
+    const api = client.nodeApi;
+    const submittable = api.tx.vote.vote(vote.voteId, myVote === "Yes");
+    await signer.signAndSend({
+        signerId: client.currentAddress!,
+        submittable,
+        callback
+    });
 
-    // TODO sign and submit extrinsic
     return new Promise<Vote>((resolve, reject) => {
```

Comment:
You must remove this promise with the time-out, it is part of the mock. However, you should return the updated vote so that the UI is immediately showing the right information.